nav2_behavior_tree/plugins/condition/goal_reached_condition.cpp

```diff
@@ -73,17 +73,18 @@ void GoalReachedCondition::initialize()
 
 bool GoalReachedCondition::isGoalReached()
 {
-  geometry_msgs::msg::PoseStamped current_pose;
+  geometry_msgs::msg::PoseStamped goal;
+  getInput("goal", goal);
+  if (goal.header.frame_id.empty()) goal.header.frame_id = global_frame_;
```

Comment:
Lets make a variable `std::string global_frame` that we populate in this if condition and then use in the `getCurrentPose` method. Lets not overwrite things that we don't expect the node to do

Reply:
Also properly format this, don't use one line.

Reply:
^ Actually I wonder if we should even bother. If a goal header isn't set, wouldn't alot of other things also have errors and fail at various stages? Maybe we can actually just delete the global_frame port completely.